bin/MorphingMSSMvsSM.cpp

```diff
@@ -516,47 +518,113 @@ int main(int argc, char **argv) {
   }
   else if(categorization == "sm-ml-only"){
     cats["et"] = {
-      { 1, "et_xxh"}, // SM Signal Category
-
       {13, "et_tt"},
       {15, "et_zll"},
       {16, "et_misc"},
       {20, "et_emb"},
       {21, "et_ff"}
     };
+    if(split_sm_signal_cat){
+      sm_signal = {
+        // Split SM Signal Categories
+        { 1, "et_xxh_bin_1"},
+        { 3, "et_xxh_bin_2"},
+        { 4, "et_xxh_bin_3"},
+        { 5, "et_xxh_bin_4"},
+        { 6, "et_xxh_bin_5"},
+        { 7, "et_xxh_bin_6"}
+      };
+    }
+    else{
+      sm_signal = {
+        // 2D SM Signal Category
+        { 1, "et_xxh"}
+      };
+    }
+    cats["et"].reserve(cats["et"].size() + std::distance(sm_signal.begin(), sm_signal.end()));
```

Comment:
Just for my education: Is this necessary or is there an advantage in reserving the space beforehand? Maybe because then only one allocation is performed or so?

Reply:
on stackoverflow the said this is a "good practiice" to do so :D